### PR TITLE
implement top site abtesting

### LINKED ISSUE #4871: [AB Test] Top Site, Group setting - Default Top Site Update 2/3
Spec check notes:
doc:https://docs.google.com/document/d/1z9I9zlBM05OKwtALps4oYa96VPbYg4NZKTrLRQ0TApI/edit#
country - sim card ; new user- check from version

Group method:
https://docs.google.com/document/d/1z9I9zlBM05OKwtALps4oYa96VPbYg4NZKTrLRQ0TApI/edit#heading=h.g3d4s4ch64r2

Data of   str_top_sites_fixed_items ; str_top_sites_default_items 
https://docs.google.com/document/d/1z9I9zlBM05OKwtALps4oYa96VPbYg4NZKTrLRQ0TApI/edit#bookmark=id.zt707ssinou

Metrics: https://docs.google.com/spreadsheets/d/1T0d8HTCnhA495zsaszSAAAqz_PRISAtlyTBnOCfig2U/edit#gid=1007897701

Tracking even add two new parameter: source and position

Also need to do satisfaction survey for each group
Need to make sure: can send push notification for different group to do satisfaction survey

### PR DESCRIPTION
**Pull request by proevan**
to fix #4871

### CODE REVIEW

File: app/src/main/res/raw-mcc510/abtesting.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
       {
         "experiment_name": "smart_shopping_copy_a",
         "bucket_range_start": 1,
-        "bucket_range_end": 7
+        "bucket_range_end": 4
```
Comment (benjamin-cheng) [inline app/src/main/res/raw-mcc510/abtesting.json:10]: Do you accidentally change the value of smart shopping search?

File: app/src/main/res/raw-mcc510/abtesting.json [hunk 2]
```diff
@@ -20,5 +20,37 @@
...
+  },
+  {
+    "name": "ab_testing_experiment_name_top_sites",
+    "enabled": true,
+    "newUserOnly": true,
+    "buckets": [
+      {
+        "experiment_name": "id_topsite_rr_0429_control",
+        "bucket_range_start": 1,
+        "bucket_range_end": 4
```
Comment (benjamin-cheng) [inline app/src/main/res/raw-mcc510/abtesting.json:32]: Just want to confirm do we want different bucket range setting in Indonesia?

File: app/src/main/java/org/mozilla/rocket/home/HomeViewModel.kt [hunk 3]
```diff
@@ -266,12 +266,12 @@ class HomeViewModel(
...
     }
 
-    fun onRemoveTopSiteClicked(site: Site) = viewModelScope.launch {
+    fun onRemoveTopSiteClicked(site: Site, position: Int) = viewModelScope.launch {
         when (site) {
             is Site.UrlSite.RemovableSite -> {
                 removeTopSiteUseCase(site)
                 updateTopSitesData()
-                TelemetryWrapper.removeTopSite(site.isDefault)
+                TelemetryWrapper.removeTopSite(site.isDefault, position, site.title)
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/rocket/home/HomeViewModel.kt:274]: Do we also log the user top site title?

File: app/src/main/res/raw-mcc510/abtesting.json (hunk 1 above)
Comment (proevan) [reply app/src/main/res/raw-mcc510/abtesting.json:10]: yes XD

File: app/src/main/res/raw-mcc510/abtesting.json (hunk 2 above)
Comment (proevan) [reply app/src/main/res/raw-mcc510/abtesting.json:32]: yes we want different bucket range for india and indonesia
https://github.com/mozilla-tw/FirefoxLite/issues/4871#issuecomment-611850019

**Pull request merged**